### PR TITLE
WIP: Credits screen

### PR DESCRIPTION
**Pull request by godouuu**
# Description

This PR adds a CreditsScreen PackedScene and hooks up to the MainMenuUiManager. The credits screen uses a similar UI to a level screen but only has the back, settings, and exit button. The credits text is built mostly with rich text labels in box containers. Graphic design isn't exactly my passion so I could probably use some feedback on how the credits screen looks. It's pretty basic as it is now.

## Related issue(s)

Related to https://github.com/Saplings-Projects/Birthday-2024/issues/12

## List of changes

A more detailed list of the changes.
- Renamed the credits.tscn to CreditsScreen and moved it to the ScenePrefabs folder.
- Made a CreditsScreen.gd script for button controls
- Added the CreditsScreen to the MainMenuUiManager

## Additional notes

This branch was built on top of the fade-transition in my forked repo so that fade-transition PR I've got open would probably need to be merged before this one.

Also here is a screenshot of the credits screen as of now
image

### CODE REVIEW

**Comment by Mikotey** [comment]
oops oops!
First time someone has used a draft PR. Though the "Ready for review" button was how I was supposed to start a review since it was requested

**Comment by Mikotey** [comment]
That should be fine. The credits screen doesn't need to be fancy. But I dunno if I can give much feedback on that end. It might be better to ask HDAK in project chat what he thinks

image
You should apply the "Fauna_2024_birthday_game_theme" to the root node of the credits screen. This way it will apply the font and colors

image
You will also need to remove the theme overrides and color modulation so it won't be black and instead use the gold color

It does end up a bit hard to read, but there is a lot of white space, so you could make the text bigger. That should help thicken the font.
You can also add the names of those in the pinned message of project chat to give a proper sense of how big each section will be: https://discord.com/channels/877085190655205406/1162414864845783161/1162414864845783161

**Comment by Casulono** [comment]
couldn't do a proper review yet but we could change Patronage to e.g Game idea

**Comment by godouuu** [comment]
How it looks from these latest commits:
image

File: Birthday-2024-Project/ScenePrefabs/CreditsScreen.tscn [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+scroll_active = false
+
+[node name="Music" type="RichTextLabel" parent="HBoxContainer/RightColumn"]
+layout_mode = 2
+theme_override_colors/font_outline_color = Color(0, 0, 0, 1)
+theme_override_constants/outline_size = 1
+theme_override_font_sizes/normal_font_size = 30
+bbcode_enabled = true
+text = "[center][font_size=40]Music[/font_size]
+kessokujason
```
Comment (Casulono) [inline Birthday-2024-Project/ScenePrefabs/CreditsScreen.tscn:241]: in light of recent news, this should be "-" instead of "kessokujason"

**Review by Casulono — DISMISSED** [review]

**Comment by godouuu** [comment]
Screen with latest change
image

**Review by Mikotey — APPROVED** [review]

**Pull request merged**